### PR TITLE
CORE-14306. [ADVAPI32] Fix a Clang-Cl warning about dwError

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Assumed fix...
Unless we revert the `TRACE` use too and drop `dwError`. Would you prefer that?

JIRA issue: [CORE-14306](https://jira.reactos.org/browse/CORE-14306)

- 91b50f9ccb530e9544a224df0f0a3e5630441ab6 (@HBelusca) is the culprit.
- aba6ce5b84e34ce02b14b689c66be3cc7830e9eb (@EricKohl) made it even more obvious.

## Proposed changes

- "warning: variable 'dwError' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]"

plus

- Add an ASSERT() if no HandlerFunction* is set (i.e. if RegisterServiceCtrlHandler*() was not called).
- Fix 2 TRACE() format copypastas.

### CODE REVIEW

File: dll/win32/advapi32/service/sctrl.c [hunk 1]
```diff
@@ -517,31 +517,33 @@ static DWORD
...
-        dwError = ERROR_SUCCESS;
     }
     else if (lpService->HandlerFunctionEx)
     {
         /* FIXME: Send correct 2nd and 3rd parameters */
         (lpService->HandlerFunctionEx)(ControlPacket->dwControl,
                                        0, NULL,
                                        lpService->HandlerContext);
-        dwError = ERROR_SUCCESS;
     }
```
Comment (ThFabba) [inline dll/win32/advapi32/service/sctrl.c]: Does clang stop complaining if you just do `else ASSERT(FALSE);` or some variation thereof? That seems simpler.

Comment (SergeGautherie) [reply dll/win32/advapi32/service/sctrl.c]: > `else ASSERT(FALSE);`

As expected, that doesn't help.

**Comment by SergeGautherie** [comment]
+ a FIXME comment, + an additional commit.

File: dll/win32/advapi32/service/sctrl.c [hunk 2]
```diff
@@ -1010,6 +1015,9 @@ StartServiceCtrlDispatcherA(const SERVICE_TABLE_ENTRYA *lpServiceStartTable)
                                          lpServiceStartTable[i].lpServiceName);
         lpActiveServices[i].ServiceMain.A = lpServiceStartTable[i].lpServiceProc;
         lpActiveServices[i].hServiceStatus = NULL;
+        lpActiveServices[i].HandlerFunction = NULL;
```
Comment (ThFabba) [inline dll/win32/advapi32/service/sctrl.c]: Look at line 998/1003

Comment (SergeGautherie) [reply dll/win32/advapi32/service/sctrl.c]: Right. I missed `HEAP_ZERO_MEMORY`.
Reverted.

**Comment by SergeGautherie** [comment]
Improved additional fixes.

File: dll/win32/advapi32/service/sctrl.c [hunk 3]
```diff
@@ -517,31 +517,41 @@ static DWORD
...
                                        0, NULL,
                                        lpService->HandlerContext);
-        dwError = ERROR_SUCCESS;
+    }
+    else
+    {
+        FIXME("Service '%S' has no handler function set: dwControl=%lu ignored\n",
+              (PWSTR)((ULONG_PTR)ControlPacket + ControlPacket->dwServiceNameOffset),
+              ControlPacket->dwControl);
+        /* NotForDefaultBuild: ASSERT(lpService->HandlerFunction || lpService->HandlerFunctionEx); */
```
Comment (SergeGautherie) [inline dll/win32/advapi32/service/sctrl.c]: Do you think this `ASSERT` could be enabled by default?

File: dll/win32/advapi32/service/sctrl.c [hunk 4]
```diff
@@ -517,31 +517,41 @@ static DWORD
...
 {
-    DWORD dwError;
+    DWORD dwError = ERROR_SUCCESS;
 
     TRACE("ScControlService(%p %p)\n",
           lpService, ControlPacket);
 
     if (lpService == NULL || ControlPacket == NULL)
-        return ERROR_INVALID_PARAMETER;
+    {
```
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply dll/win32/advapi32/service/sctrl.c]: Then, let's fully revert `dwError`.

File: dll/win32/advapi32/service/sctrl.c (hunk 3 above)
Comment (SergeGautherie) [reply dll/win32/advapi32/service/sctrl.c]: Simple `ASSERT()` then.
Though unexpected, it is not obvious to me why this case would be impossible.

File: dll/win32/advapi32/service/sctrl.c [hunk 5]
```diff
@@ -704,6 +714,7 @@ RegisterServiceCtrlHandlerW(LPCWSTR lpServiceName,
 
     Service->HandlerFunction = lpHandlerProc;
     Service->HandlerFunctionEx = NULL;
+    Service->HandlerContext = NULL;
```
[the remark this answers is not in the document]
Comment (SergeGautherie) [reply dll/win32/advapi32/service/sctrl.c]: Reverted.

File: dll/win32/advapi32/service/sctrl.c [hunk 6]
```diff
@@ -991,25 +1003,24 @@ StartServiceCtrlDispatcherA(const SERVICE_TABLE_ENTRYA *lpServiceStartTable)
...
     }
 
-    dwActiveServiceCount = i;
-
     /* Allocate the service table */
     lpActiveServices = RtlAllocateHeap(RtlGetProcessHeap(),
                                        HEAP_ZERO_MEMORY,
-                                       dwActiveServiceCount * sizeof(ACTIVE_SERVICE));
+                                       i * sizeof(ACTIVE_SERVICE));
     if (lpActiveServices == NULL)
```
Comment (SergeGautherie) [inline dll/win32/advapi32/service/sctrl.c]: it is not obvious to me what prevents a later NULL-deref of `lpActiveServices`, as `dwActiveServiceCount` remains set.

**Comment by SergeGautherie** [comment]
Superseded by db60cd219c6c6f4443d9f7cb4c708f690662a2f0.

**Pull request closed without merging**